components/patient-data/impl/src/main/java/org/phenotips/data/internal/controller/AbstractSimpleController.java

```diff
@@ -85,7 +91,27 @@
     @Override
     public void save(Patient patient)
     {
-        throw new UnsupportedOperationException();
+        try {
+            XWikiDocument doc = (XWikiDocument) this.documentAccessBridge.getDocument(patient.getDocument());
+            BaseObject xwikiDataObject = doc.getXObject(Patient.CLASS_REFERENCE);
+            if (xwikiDataObject == null) {
+                throw new NullPointerException(ERROR_MESSAGE_NO_PATIENT_CLASS);
```

Comment:
Better than this would be `IlegalArgumentException`.

But I wonder if a custom exception would be better (`InvalidPatientException`), since I don't like that `ERROR_MESSAGE_NO_PATIENT_CLASS`.